lib/tree.js

```diff
@@ -7,18 +7,36 @@ var metadata = require('./helpers/metadata');
 function tree(opts) {
   opts = _.defaults(opts || {}, {
     deep: false,
+    depth: null,
   });
 
   var tasks = this._registry.tasks();
-  var nodes = _.map(tasks, function(task) {
-    var meta = metadata.get(task);
 
-    if (opts.deep) {
-      return meta.tree;
-    }
-
-    return meta.tree.label;
-  });
+  var nodes;
+  if (!opts.deep) {
+    nodes = _.map(tasks, function(task) {
+      var meta = metadata.get(task);
+      return meta ? meta.tree.label : '';
+    });
+  } else {
+    var maxDepth = (opts.depth == null || opts.depth > 0) ? opts.depth : 1;
+    nodes = _.map(this._registry.tasks(), function(task) {
+      var node = {};
+      var meta = metadata.get(task);
+      if (!meta) {
+        return node;
+      }
+      if (meta.orig) {
+        if (meta.orig.description) {
```

Comment:
can these be retrieved inside gulp-cli?  were you just trying to reduce the times we iterate over the tasks list?  I'd prefer `description` and `flags` not to mean anything to undertaker since they are a very gulp-cli specific thing